src/Microsoft.AspNet.Mvc.ModelBinding/FileUpload/FormFileExtensions.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.AspNet.Http;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Extension methods for <see cref="IFormFile"/>.
+    /// </summary>
+    public static class FormFileExtensions
+    {
+        /// <summary>
+        /// Saves the contents of an uploaded file.
+        /// </summary>
+        /// <param name="formFile">The <see cref="IFormFile"/>.</param>
+        /// <param name="filename">The name of the file to create.</param>
+        public static void SaveAs([NotNull] this IFormFile formFile, string filename)
```

Comment:
@ajaybhargavb please come to a consensus on where this is going :+1:. M.A.H.E is good if we can keep the experience the same - no extra dependencies that MVC doesn't already have, no extra namespaces other than M.A.H and M.A.M.